src/group_theory/subgroup.lean

```diff
@@ -1268,38 +1303,58 @@ begin
 end
 
 @[to_additive]
-lemma map_comap_eq_self {f : G →* H} {K : subgroup H} (h : K ≤ f.range) :
-  map f (comap f K) = K :=
+lemma map_comap_eq_self {f : G →* N} {H : subgroup N} (h : H ≤ f.range) :
+  map f (comap f H) = H :=
 by rwa [map_comap_eq, inf_eq_right]
 
 @[to_additive]
-lemma map_comap_eq_self_of_surjective {f : G →* H} (h : function.surjective f) (K : subgroup H) :
-  map f (comap f K) = K :=
+lemma map_comap_eq_self_of_surjective {f : G →* N} (h : function.surjective f) (H : subgroup N) :
+  map f (comap f H) = H :=
 map_comap_eq_self ((range_top_of_surjective _ h).symm ▸ le_top)
 
 @[to_additive]
-lemma comap_injective {f : G →* H} (h : function.surjective f) : function.injective (comap f) :=
+lemma comap_injective {f : G →* N} (h : function.surjective f) : function.injective (comap f) :=
 λ K L hKL, by { apply_fun map f at hKL, simpa [map_comap_eq_self_of_surjective h] using hKL }
 
 @[to_additive]
-lemma comap_map_eq_self {f : G →* H} {K : subgroup G} (h : f.ker ≤ K) :
-  comap f (map f K) = K :=
+lemma comap_map_eq_self {f : G →* N} {H : subgroup G} (h : f.ker ≤ H) :
+  comap f (map f H) = H :=
 by rwa [comap_map_eq, sup_eq_left]
 
 @[to_additive]
-lemma comap_map_eq_self_of_injective {f : G →* H} (h : function.injective f) (K : subgroup G) :
-  comap f (map f K) = K :=
+lemma comap_map_eq_self_of_injective {f : G →* N} (h : function.injective f) (H : subgroup G) :
+  comap f (map f H) = H :=
 comap_map_eq_self (((ker_eq_bot_iff _).mpr h).symm ▸ bot_le)
 
 @[to_additive]
-lemma map_injective {f : G →* H} (h : function.injective f) : function.injective (map f) :=
+lemma map_injective {f : G →* N} (h : function.injective f) : function.injective (map f) :=
 λ K L hKL, by { apply_fun comap f at hKL, simpa [comap_map_eq_self_of_injective h] using hKL }
 
 @[to_additive]
-lemma map_eq_comap_of_inverse {f : G →* H} {g : H →* G} (hl : function.left_inverse g f)
-  (hr : function.right_inverse g f) (K : subgroup G) : map f K = comap g K :=
+lemma map_eq_comap_of_inverse {f : G →* N} {g : N →* G} (hl : function.left_inverse g f)
+  (hr : function.right_inverse g f) (H : subgroup G) : map f H = comap g H :=
 set_like.ext' $ by rw [coe_map, coe_comap, set.image_eq_preimage_of_inverse hl hr]
 
+/-- Given `f(A) = f(B)`, `ker f ≤ A`, and `ker f ≤ B`, deduce that `A = B`  -/
+@[to_additive] lemma eq_of_map_eq_le_ker
+  (H K : subgroup G) (hf : map f H = map f K) (hH : f.ker ≤ H) (hK : f.ker ≤ K) :
```

Comment:
`H` and `K` (and `f`) can be implicit since they can be inferred from `hf`, `hH` and `hK`. In names, we separate parameters with `of` and put left arguments of relations before right arguments:
```suggestion
@[to_additive] lemma eq_of_map_eq_of_ker_le
  {H K : subgroup G} (hf : map f H = map f K) (hH : f.ker ≤ H) (hK : f.ker ≤ K) :
```
You might even want to draw the connection to injectivity by rearranging the parameters a bit:
```suggestion
@[to_additive] lemma map_inj_of_le_ker
  {H K : subgroup G} (hH : f.ker ≤ H) (hK : f.ker ≤ K) (hf : map f H = map f K) :
```

Reply:
(Do we already have that `function.injective f` implies `function.injective (map f)`? That would be a nice corollary of this result if it's not there yet.)